hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip551/contracts/AtomicBatchEndToEndSmartContractsHTSCallsAndAssociationsTest.java

```diff
@@ -0,0 +1,2237 @@
+                                // Mint NFT
+                                contractCall(
+                                                HTS_CALLS_CONTRACT,
+                                                "mintTokenCall",
+                                                nonFungibleAddress.get(),
+                                                BigInteger.valueOf(0L),
+                                                new byte[][] {
+                                                    "nft1".getBytes(),
+                                                    "nft2".getBytes(),
+                                                    "nft3".getBytes(),
+                                                    "nft4".getBytes()
+                                                })
+                                        .payingWith(OWNER)
+                                        .gas(GAS_TO_OFFER)
+                                        .via("callNftMintContractInnerTxn"),
+                                // Associate FT to the associate contract
+                                contractCall(ASSOCIATE_CONTRACT, "associateTokenToThisContract", fungibleAddress.get())
+                                        .payingWith(OWNER)
+                                        .gas(GAS_TO_OFFER)
+                                        .via("associateFungibleTokenContractInnerTxn"),
+                                // Associate NFT to the associate contract
+                                contractCall(
+                                                ASSOCIATE_CONTRACT,
+                                                "associateTokenToThisContract",
+                                                nonFungibleAddress.get())
+                                        .payingWith(OWNER)
+                                        .gas(GAS_TO_OFFER)
+                                        .via("associateNonFungibleTokenContractInnerTxn"))
+                        .payingWith(DEFAULT_BATCH_OPERATOR)
+                        .via("atomicBatchTxn")
+                        .hasKnownStatus(SUCCESS)),
+
+                // Assert the tokens supply is updated correctly
+                getTokenInfo(FT_TOKEN).hasTotalSupply(10L),
+                getTokenInfo(NON_FUNGIBLE_TOKEN).hasTotalSupply(4L),
+                // Assert the owner account has no tokens
+                getAccountBalance(OWNER).hasTokenBalance(FT_TOKEN, 0L).hasTokenBalance(NON_FUNGIBLE_TOKEN, 0L),
+                // Assert the receiver associated account does not have tokens but is associated with them
+                getAccountBalance(ASSOCIATE_CONTRACT)
+                        .hasTokenBalance(FT_TOKEN, 0L)
+                        .hasTokenBalance(NON_FUNGIBLE_TOKEN, 0L),
+                getAccountInfo(ASSOCIATE_CONTRACT)
+                        .hasToken(relationshipWith(FT_TOKEN))
+                        .hasToken(relationshipWith(NON_FUNGIBLE_TOKEN)),
```

Comment:
This is implied by `getAccountBalance(ASSOCIATE_CONTRACT).hasTokenBalance(...)` on the lines immediately above